cmd/acr/purge.go

```diff
@@ -614,6 +621,53 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 	return deletedTagsCount, deletedManifestsCount, nil
 }
 
+func filterManifests(manifest acr.ManifestAttributesBase, cm customManifest, doNotDelete Set[string], candidatesToDelete []acr.ManifestAttributesBase) ([]acr.ManifestAttributesBase, Set[string], error) {
+	switch *manifest.MediaType {
+	case MediaTypeManifestList:
+		if manifest.Tags == nil {
+			// If the manifest has no tag, it is a candidate for deletion
+			candidatesToDelete = append(candidatesToDelete, manifest)
+		} else {
+			// If this multiarchitecture manifest has tag, its dependent
+			// manifests are maked to not be deleted
+			for _, dependentManifest := range cm.Manifests {
+				doNotDelete.Add(dependentManifest.Digest)
+			}
+		}
+	case MediaTypeArtifactManifest, MediaTypeImageManifest:
+		if manifest.Tags == nil {
+			if cm.Subject != nil {
+				// if manifest has subject, manifest is marked to not be deleted
```

Comment:
You may refactor the nested if structures

Reply:
e.g. 
```suggestion
		if manifest.Tags != nil {
			break
		}
		if cm.Subject != nil {
			// if manifest has subject, manifest is marked to not be deleted
			doNotDelete.Add(*manifest.Digest)
			break
		}
		// If the manifest has no tag or subject, it is a candidate for deletion
		candidatesToDelete = append(candidatesToDelete, manifest)
```

Reply:
changed two case statements except the MediaTypeImageIndex, since we have different operation for both `Tags == nil` and `Tags != nil`. Within `Tags == nil`, it is necessary to determine if `subject == nil` or not.